make check error

When I run `make check` I get the following error:

```bash
stdin_reader.cc: In member function ‘virtual std::vector<std::__cxx11::basic_string<char> > scheduler_simulator::StdinReader::next()’:
stdin_reader.cc:29:8: error: ignoring return value of ‘char* fgets(char*, int, FILE*)’, declared with attribute warn_unused_result [-Werror=unused-result]
   fgets(line, SCHEDULER_SIMULATOR_INPUT_LINE_LENGTH, stdin);
   ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/.github/workflows/c-cpp.yml
+++ b/.github/workflows/c-cpp.yml
@@ -20,6 +20,9 @@ jobs:
     - name: make
       run: make
     - name: make check
-      run: make check
+      run: |
+        make check || ret=$?
+        cat ./test-suite.log
+        $(exit ${ret})
     - name: make distcheck
       run: make distcheck

--- a/stdin_reader.cc
+++ b/stdin_reader.cc
@@ -1,38 +1,24 @@
 #include "stdin_reader.hh"
+
+#include <iostream>
+#include <iterator>
 #include <sstream>
-#include <stdio.h>
-#include <string.h>
 #include <string>
 
 using namespace scheduler_simulator;
-using namespace std;
 
-StdinReader::StdinReader() {
-  fseek(stdin, 0L, SEEK_SET);
-}
+StdinReader::StdinReader() { }
 
-//! Interesting method to find if we 
-//! are about to reach EOF
+//! Interesting method to find if we are about to reach EOF
 bool StdinReader::is_next() {
-  char c = fgetc(stdin);
-  if (c == EOF)
-    return false;
-  else {
-    ungetc(c, stdin);
-    return true;
-  }
+  return EOF != std::cin.peek();
 }
 
 //! Read the next line from stdin
 std::vector<std::string> StdinReader::next() {
-  char line [SCHEDULER_SIMULATOR_INPUT_LINE_LENGTH];
-  fgets(line, SCHEDULER_SIMULATOR_INPUT_LINE_LENGTH, stdin);
-  stringstream ss(line);
+  std::stringstream s;
+  std::cin.get(*s.rdbuf()).ignore();
 
-  vector<string> output;
-  string token;
-  while (ss >> token)
-    output.push_back(token);
-
-  return output;
+  return {std::istream_iterator<std::string>(s),
+    std::istream_iterator<std::string>()};
 }

--- a/tests/integration-test-runner.py
+++ b/tests/integration-test-runner.py
@@ -1,7 +1,7 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import yaml
-import subprocess 
+import subprocess
 import os
 import sys
 
@@ -10,43 +10,41 @@ TEST_CONFIG_PATH = os.environ['TEST_CONFIG']
 TEST_BINARY      = os.environ['TEST_BINARY']
 
 def die(str_):
-	print(str_)
-	sys.exit(-1)
+  print(str_)
+  sys.exit(-1)
 
 def launch_test(settings):
-	name = settings['algorithm']
-	ops = settings['parameters']
-	input_ = settings['input']
-	output_ = settings['output']
+  name = settings['algorithm']
+  ops = settings['parameters']
+  input_ = settings['input']
+  output_ = settings['output']
 
-	cmd = TEST_BINARY + " -i - " + ops
+  cmd = TEST_BINARY + " -i - " + ops
 
-	p = subprocess.Popen(cmd, shell=True, stdin=subprocess.PIPE, stdout=subprocess.PIPE);
-	p.stdin.write(input_)
-	p.stdin.close()
-	given_output = p.stdout.read()
+  p = subprocess.Popen(cmd, shell=True, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+  given_output = p.communicate(input=input_.encode())[0].decode()
 
-	test = (output_ == given_output)
-	if not test:
-		print("Code output: ")
-		print(given_output)
-		print("Reference output: ")
-		print(output_)
-	
-	return test
+  test = (output_ == given_output)
+  if not test:
+    print("Code output: ")
+    print(given_output)
+    print("Reference output: ")
+    print(output_)
+
+  return test
 
 def main():
-	with open(TEST_CONFIG_PATH, 'r') as stream:
-		try:
-			settings = yaml.load(stream)
-			for test in settings:
-				name = test['algorithm']
-				if launch_test(test) != True:
-					die("ERROR in integration test ALGORITHM: " + name)
+  with open(TEST_CONFIG_PATH, 'r') as stream:
+    try:
+      settings = yaml.safe_load(stream)
+      for test in settings:
+        name = test['algorithm']
+        if launch_test(test) != True:
+          die("ERROR in integration test ALGORITHM: " + name)
 
-		except yaml.YAMLError as exc:
-			print(exc)
+    except yaml.YAMLError as exc:
+      print(exc)
 
 
 if __name__ == "__main__":
-	main()
+  main()

